### ISSUE TITLE
MPIPreferences segfaults

### ISSUE THREAD

**Issue by luraess** [JuliaParallel/MPI.jl#560]
Running `MPIPreferences.use_system_binary()` segfaults if `using MPI` was not executed before. This behaviour is observed on MPI 0.20-dev (master) branch with Julia 1.7.1 and Julia 1.7.2 on Linux (Ubuntu 20.04).

**Comment by vchuravy** [comment]
Please post a full error trace, and what the system implementation is. I assume it is OpenMPI and this is #519 again

**Comment by luraess** [comment]
```
luraess@superzack:~/scratch/dev/MPI_master$ julia --project

(MPI) pkg> st
     Project MPI v0.20.0-dev
      Status `/scratch-1/luraess/dev/MPI_master/Project.toml`
  [ffbed154] DocStringExtensions v0.8.6
  [3da0fdf6] MPIPreferences v0.1.0
  [ae029012] Requires v1.3.0
  [7cb0a576] MPICH_jll v4.0.2+0
  [f1f71cc9] MPItrampoline_jll v3.3.1+0
  [9237b28f] MicrosoftMPI_jll v10.1.3+2
  [fe0851c0] OpenMPI_jll v4.1.2+0
  [8ba89e20] Distributed
  [8f399da3] Libdl
  [9e88b42a] Serialization
  [6462fe0b] Sockets

(MPI) pkg> instantiate

julia> using MPIPreferences

julia> MPIPreferences.use_system_binary()
Segmentation fault (core dumped)
luraess@superzack:~/scratch/dev/MPI_master$
```

**Comment by luraess** [comment]
Seems following env var is needed
```
export ZES_ENABLE_SYSMAN=1
```
as suggested in https://github.com/JuliaParallel/MPI.jl/pull/565#issuecomment-1100757740

**Comment by giordano** [comment]
I also ran into this issue the other day. I knew I had to set the environment variable, but I was wondering if we could be more proactive in showing a warning to the user. For example, we could move the ABI detection inside the `finally` branch of https://github.com/JuliaParallel/MPI.jl/blob/7cb10d56e34ec6031c94f6c6514437cefe15e178/lib/MPIPreferences/src/MPIPreferences.jl#L156-L161 and guard the `dlclose` in another `try`/`catch` block and issue a warning if `dlclose` segfaults with an OpenMPI implementation. I'm happy to do a pull request if this sounds like a good idea.
